src/components/Tables/TableFilterBar.tsx

```diff
@@ -1,70 +1,100 @@
 "use client";
 
 import React from "react";
-import styled from "styled-components";
 import FilterAccordion from "@/components/Tables/FilterAccordion";
 import Button from "@mui/material/Button";
+import { TableHeaders } from "@/components/Tables/Table";
+import { Filter } from "@/components/Tables/Filters";
+import styled from "styled-components";
+import { FilterAltOffOutlined } from "@mui/icons-material";
 
-export interface FilterText {
-    [key: string]: string;
+interface Props<Data> {
+    setFilters: (filters: Filter<Data, any>[]) => void;
+    setAdditionalFilters: (filters: Filter<Data, any>[]) => void;
+    handleClear: () => void;
+    headers: TableHeaders<Data>;
+    filters: Filter<Data, any>[];
+    additionalFilters: Filter<Data, any>[];
+    toggleableHeaders: readonly (keyof Data)[];
+    setShownHeaderKeys: (headers: (keyof Data)[]) => void;
+    shownHeaderKeys: readonly (keyof Data)[];
 }
 
-const StyledFilterAccordion = styled(FilterAccordion)`
-    width: 100%;
-    overflow: visible;
+const StyledButton = styled(Button)`
+    align-self: center;
+    justify-self: flex-end;
 `;
 
-const StyledFilterBar = styled.input`
-    height: 39px;
-    font-size: 14px;
-    width: 15rem;
+const Styling = styled.div`
+    background-color: transparent;
+    display: flex;
+    align-items: stretch;
+    flex-wrap: wrap;
+    padding: 0.5rem 0;
+    gap: 2rem;
     overflow: visible;
-    box-shadow: none;
-    padding: 4px 12px 4px 12px;
+    display: flex;
+    width: 100%;
 
-    &:focus {
-        outline: none;
+    @media (min-width: 500px) {
+        flex-wrap: nowrap;
     }
 `;
 
-interface Props {
```

Comment:
would rather the order of the file doesn't change _too_ much, so that the git diff is more readable!